File: avalon-extensions/avalon-ip2location-spring-boot-starter/src/main/java/io/github/lishangbu/avalon/ip2location/exception/InvalidIpAddressException.java
package io.github.lishangbu.avalon.ip2location.exception;

/**
 * 无效IP地址异常类
 *
 * @author lishangbu
 * @since 2025/4/12
 */
public class InvalidIpAddressException extends IpToLocationException {
  /** 构造一个新的无效IP地址异常，并设置默认错误信息 */
  public InvalidIpAddressException() {
    super("Invalid IP address.");
  }
}

File: avalon-platform/avalon-ip2location-spring-boot-starter/src/main/kotlin/io/github/lishangbu/avalon/ip2location/exception/InvalidIpAddressException.kt
package io.github.lishangbu.avalon.ip2location.exception

/**
 * 无效 IP 地址异常
 *
 * 当传入的 IP 地址格式非法时抛出
 */
class InvalidIpAddressException : IpToLocationException("Invalid IP address.")
